src/shared/concentrators/AMMConcentratorBase.sol

```diff
@@ -398,15 +401,20 @@ abstract contract AMMConcentratorBase is ReentrancyGuard, ERC4626 {
         return _underlyingAmount;
     }
 
-    /// @dev Claims all rewards for msg.sender and sends them to receiver
+    /// @dev Claims all rewards for _owner and sends them to _receiver
+    /// @param _owner - The owner of rewards
     /// @param _receiver - The recipient of rewards
-    /// @return _rewards - The amount of compounder shares sent to the _receiver
-    function claim(address _receiver) public nonReentrant returns (uint256 _rewards) {
+    /// @return _rewards - The amount of Compounder shares sent to the _receiver
+    function claim(address _owner, address _receiver) public nonReentrant returns (uint256 _rewards) {
         if (settings.pauseClaim) revert ClaimPaused();
+        
+        if (msg.sender != multiClaimer) {
+            _owner = msg.sender;
+        }
```

Comment:
I see that `claim` is being called via the zero address in several cases, which then triggers the body of the condition here. Would it make more sense to not have this condition and instead simply call `claim` using the message sender directly? Could maybe even safe gas. Happy to understand better what purpose this condition serves other than address defaulting. 

Also, is there any case in which `claim` is not called using zero address? Do we integrate `claim` via the frontend later?

Reply:
Context for this change is that we need to be able to claim from multiple Concentrator vaults in 1 transaction, see https://github.com/FortressFinance/issues/issues/119.

One way to implement it is to enable the specification of the `owner` when calling the `claim` function - we can then call `claim` for a user from multiple vaults. Since it's a UX feature we need internally from the Concentrator vaults (as far as I can think of), I rather gate (whitelist) this feature to minimize attack surface.

Note that the contracts can still be turned into "autonomous" of we disable the `owner` (assign an `address(0)` to it).